src/FeaturesViewEngine/ControllerFeaturesViewEngine.cs

```diff
@@ -57,7 +58,8 @@ public virtual string NamespacePrefixToRemove(ControllerContext controllerContex
         {
             var featurePath = GetFeaturePath(controllerContext);
             var controllerName = GetControllerName(controllerContext);
-            var cacheKey = CreateCacheKey(featurePath, viewName, controllerName);
+            var displayModes = GetAvailableDisplayModes(controllerContext);
```

Comment:
Method name should clearly indicate that not all the registered display modes will be returned, but only those that match current request (context).